cmd/hdfs/touch.go

```diff
@@ -25,18 +26,25 @@ func touch(paths []string, noCreate bool) {
 			fatal(&os.PathError{"mkdir", p, os.ErrNotExist})
 		}
 
-		_, err := client.Stat(p)
+		finfo, err := client.Stat(p)
 		exists := !os.IsNotExist(err)
 		if (err != nil && exists) || (!exists && noCreate) {
 			fatal(err)
 		}
 
 		if exists {
-			now := time.Now()
-			mtime := now
-			atime := now
-
-			err = client.Chtimes(p, mtime, atime)
+			if accessTime {
+				now := time.Now()
+				atime := now
+				err = client.Chtimes(p, atime, finfo.ModTime())
```

Comment:
I think this error can be hidden in the case of `-am` (both flags). There should just be one `Chtimes` call at the end.